### PR TITLE
[improve][pip] PIP-347: Add role field in consumer's stat

### PR DESCRIPTION
**Pull request by thetumbled**
### Motivation

By setting config `allowAutoSubscriptionCreation=true`, we allow user to create subscriptions automatically if user have the `consume` permission. 
We have a strong need to trace back the owner of a specific subscription, but currently there is no such way to achieve that.

### Modifications

Add field `appId` in the stats response, which contains role info when cluster use JWT.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/thetumbled/pulsar/pull/48

### CODE REVIEW

**Comment by thetumbled** [comment]
PTAL, thanks. @lhotari @codelipenghui @BewareMyPower @Technoboy-

**Review by BewareMyPower — DISMISSED** [review]

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/ConsumerStats.java [hunk 1]
```diff
@@ -25,6 +25,9 @@
  * Consumer statistics.
  */
 public interface ConsumerStats {
+    /** the role used for authorization. */
+    String getRole();
```
[the remark this answers is not in the document]
Comment (thetumbled) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/ConsumerStats.java:29]: this is the pip for this pr: https://github.com/apache/pulsar/pull/22564.

**Review by dao-jun — APPROVED** [review]

**Comment by dao-jun** [comment]
@thetumbled Please resolve conflicts

**Comment by thetumbled** [comment]
> @thetumbled Please resolve conflicts

Done, thank you.

**Comment by thetumbled** [comment]
Could you help to review? thank you. @lhotari @Technoboy- @codelipenghui

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**